### PR TITLE
Tweak: Add to the GoPro link URL from the Kit Library more UTM parameters [ED-7745]

### PR DESCRIPTION
**Pull request by davseve**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: core/app/modules/kit-library/assets/js/components/item-header.js [hunk 1]
```diff
@@ -39,6 +39,19 @@ function useKitCallToActionButton( model, { apply, isApplyLoading, onConnect } )
 		}
 
 		if ( type === TYPE_PROMOTION && subscriptionPlan ) {
+			const buildSubscriptionLink = () => {
+				let url = subscriptionPlan.promotion_url;
+				if ( model.title ) {
+					// Remove special characters, replace spaces with '-' and convert url kit name to lowercase.
+					let cleanTitle = model.title.replace( /[^\w\s]/g, '' ).replace( /\s/g, '-' ).toLowerCase();
```
Comment (mark-elementor) [inline core/app/modules/kit-library/assets/js/components/item-header.js:46]: - Should use `const` instead.
- What if the title already includes a dash? Would not it be better to switch the replaces?
`const cleanTitle = model.title.replace( /\s/g, '-' ).replace( /[^\w\s-]/g, '' ).toLowerCase();`

File: core/app/modules/kit-library/assets/js/components/item-header.js [hunk 2]
```diff
@@ -39,6 +39,19 @@ function useKitCallToActionButton( model, { apply, isApplyLoading, onConnect } )
 		}
 
 		if ( type === TYPE_PROMOTION && subscriptionPlan ) {
+			const buildSubscriptionLink = () => {
+				let url = subscriptionPlan.promotion_url;
+				if ( model.title ) {
```
Comment (mark-elementor) [inline core/app/modules/kit-library/assets/js/components/item-header.js:44]: - I would add linebreaks before and after the `if` s for better readability.
- Is there a chance there will be no model.title or no model.id?

File: core/app/modules/kit-library/assets/js/components/item-header.js [hunk 3]
```diff
@@ -47,7 +60,7 @@ function useKitCallToActionButton( model, { apply, isApplyLoading, onConnect } )
 				variant: 'contained',
 				color: 'cta',
 				size: 'sm',
-				url: subscriptionPlan.promotion_url,
+				url: buildSubscriptionLink(),
```
Comment (mark-elementor) [inline core/app/modules/kit-library/assets/js/components/item-header.js:63]: I would call it something like "getButtonURL" instead.

File: core/common/modules/connect/module.php [hunk 4]
```diff
@@ -199,12 +199,12 @@ public function get_subscription_plans( $context = '' ) {
 			],
 			static::ACCESS_LEVEL_PRO => [
 				'label' => 'Pro',
-				'promotion_url' => "https://elementor.com/pro/?utm_source={$context}&utm_medium=wp-dash&utm_campaign=gopro",
+				'promotion_url' => "https://elementor.com/pro/?utm_source=$context&utm_medium=wp-dash&utm_campaign=gopro",
```
Comment (mark-elementor) [inline core/common/modules/connect/module.php:202]: Second thought on this - the variable can easily be mistaken for a part of the string here.
So for nobody to do a mistake here again, maybe it is better to do:
`'https://elementor.com/pro/?utm_source=' . $context . '&utm_medium=wp-dash&utm_campaign=gopro'`

File: core/common/modules/connect/module.php [hunk 5]
```diff
@@ -199,12 +199,12 @@ public function get_subscription_plans( $context = '' ) {
...
 				'color' => '#92003B',
 			],
 			static::ACCESS_LEVEL_EXPERT => [
 				'label' => 'Expert',
-				'promotion_url' => "https://elementor.com/pro/?utm_source={$context}&utm_medium=wp-dash&utm_campaign=goexpert",
+				'promotion_url' => "https://elementor.com/pro/?utm_source=$context&utm_medium=wp-dash&utm_campaign=goexpert",
```
Comment (mark-elementor) [inline core/common/modules/connect/module.php:207]: Same as above.

**Review by mark-elementor — CHANGES_REQUESTED** [review]

**Review by mark-elementor — APPROVED** [review]

**Review by yossi-elementor — APPROVED** [review]

**Pull request merged**